gcd/src/main/java/org/spine3/server/storage/datastore/DsEventStorage.java

```diff
@@ -116,33 +134,61 @@ public DsEventStorage(DatastoreWrapper datastore, boolean multitenant) {
 
     @Override
     public Iterator<Event> iterator(EventStreamQuery eventStreamQuery) {
-        final Query<Entity> query = toTimestampQuery(eventStreamQuery);
+        final Iterable<Query<Entity>> queries = toEventIteratorQueries(eventStreamQuery);
 
-        final Collection<Entity> entities = datastore.read(query);
+        final Collection<Entity> entities = new LinkedList<>();
+        for (Query<Entity> query : queries) {
+            final Collection<Entity> queryResults = datastore.read(query);
+            entities.addAll(queryResults);
+        }
         // Transform and filter order does not matter since both operations are performed lazily
         Collection<EventStorageRecord> events = Collections2.transform(entities, ENTITY_TO_EVENT_RECORD);
         events = Collections2.filter(events, eventPredicate(eventStreamQuery));
+        final List<EventStorageRecord> sortedEvents = new ArrayList<>(events);
 
-        final Iterator<EventStorageRecord> iterator = events.iterator();
+        // Merge sort
```

Comment:
The comment is unclear. With this detail level, it can be removed.

Please either extend it or remove.